Ex1/week6_3.py

```diff
@@ -0,0 +1,186 @@
+    """
+    try:
+        return float(my_string)
+    except ValueError:
+        return False
+
+
+def organize_closet(list_closet):
+    """
+    Gets a list of strings representing sizes of clothes and returns a list of the square root of the strings
+    that represent a number.
+    Possible input: ['Area51', '303', '2038', 'f00b4r', '314.1']
+    The output: [17.41, 45.14, 17.72]
+    :param list_closet: A list of strings representing sizes of clothes
+    :return: A list of the square root of the strings that represent a number (after rounding to 2 nums after point)
+    """
+    return [round(math.sqrt(float(size_closet)), 2) for size_closet in list_closet if size_closet.isnumeric()
+            or is_float(size_closet)]
+
+
+dice_options_gen_expression = (
+    (die1, die2, die3)
+    for die1 in range(1, 7)
+    for die2 in range(1, 7)
+    for die3 in range(1, 7)
+)
+
+
+def dice_generator_function():
+    for die1 in range(1, 7):
+        for die2 in range(1, 7):
+            for die3 in range(1, 7):
+                yield die1, die2, die3
+
+
+# Be in peace
+
+
+def words_length(sentence):
+    """
+    Function that get a sentence of words and returns a list of the length of the words in it
+    :param sentence: The sentence
+    :return: A list of the length of the words in the sentence sent
+    """
+    return [len(x) for x in sentence.split()]
+
+
+# א אוהל, פה זה פייתון
+
+
+def get_letters():
+    """
+    A function that returns a list of all lowercase/uppercase letters
+    :return: A list of all lowercase/uppercase letters
+    """
+    return [chr(letter_low) for letter_low in range(ord('a'), ord('z')+1)] +\
+           [chr(letter_up) for letter_up in range(ord('A'), ord('Z')+1)]
+
+
+# Long cat is long
+
+
+def count_words(txt):
+    """
+    The function gets a text, first cuts all characters that are not letters or whitespaces, then returns
+    a dictionary of the words (as keys) and their length (as value)
+    :param txt: The text sent
+    :return: A dictionary of the words (as keys) and their length (as value)
+    """
+    my_clean_text = ''.join(ch for ch in txt if ch.isalnum() or ch.isspace())
+    return {word: len(word) for word in my_clean_text.split()}
+
+
+def full_names(firsts_names, lasts_names, min_length=0):
+    """
+    Gets a list of first names and a list of last names and an optional parameter min_length and return
+    a list of all combinations of first names with last names that are greater or equal to the min_length
+    if passed (by default min_length is 0), that after will turn each first/last name to uppercase at
+    the first letter
+    :param firsts_names: A list of first names
+    :param lasts_names: A list of last names
+    :param min_length: (optional) The full names that are greater than this parameter
+    :return: List of all full names
+    """
+    return [first[0].upper() + first[1:] + " " + last[0].upper() + last[1:]
```

Comment:
And this code is unreadable, try to make it clearer, even if its at the cost of making it a bit longer